### ISSUE TITLE
Missing reference times in CF output

### ISSUE THREAD

**Issue by infotroph** [PecanProject/pecan#1347]
Several functions in `data.atmosphere` produce invalid CF output because they give time in units of "sec", with no internal indication what time the file started. 

## Description

```
cd ~/github_forks/pecan/modules/data.atmosphere/R
grep -R '"sec"' .
./debias.met.R:  time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:reso_len) *
./download.CRUNCEP_Global.R:    time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:ntime) * 21600,
./download.GFDL.R:    time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:2920) * 10800,
./download.GLDAS.R:                      units = "sec",
./download.MACA.R:    time <- ncdf4::ncdim_def(name='time', units="sec", vals=(1:365)*86400, create_dimvar=TRUE, unlim=TRUE)
./download.MsTMIP_NARR.R:                      units = "sec",
./download.NLDAS.R:    time <- ncdf4::ncdim_def(name = "time", units = "sec",
./met_temporal_downscale.Gaussian_ensemble.R:    time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = seq_len(reso_len) *
./upscale_met.R:  time <- ncdf4::ncdim_def(name = "time", units = "sec", vals = (1:reso_len) *
```

## Context
The CF format conventions [require](http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#time-coordinate) that time variables always include a reference time in their units attribute, and the PEcAn documentation at least implies that times should be [days since 1700-01-01 00:00:00 UTC](https://pecan.gitbooks.io/pecan-documentation/content/developers_guide/Adding-an-Input-Converter.html). Enforcing this at the CF conversion stage would simplify the logic of all the met2model.MODEL functions, and would be more reliable than assuming filenames contain start dates.

## Possible Implementation

**Comment by infotroph** [comment]
An issue to be aware of while fixing this: If adding or subtracting large numbers of seconds to calculate reference time with something like `lubridate::ymd("1700-01-01") + lubridate::seconds(x$dim$time$val)`, look out for [integer overflows in the date math](https://github.com/hadley/lubridate/issues/534).

**Comment by ankurdesai** [comment]
@jsimkins2 I think this is mostly in your functions. Can you take a look?

**Comment by dlebauer** [comment]
I think this is fixed for CRUNCEP in #1357. Could code from that PR be reused, or generalized and applied to all of the met2CF functions?

**Comment by jsimkins2** [comment]
@dlebauer Yes, that can be reused and applied to other data.atmosphere functions. The only difference is the reference date. This varies per source, download.MACA references 1900-01-01, while download.GFDL references 1861-01-01

**Comment by infotroph** [comment]
Unstaling: Still needs to be done. Shouldn't be too hard, just need to make sure we fix all instances.
